Services/LTI/classes/Screen/LtiViewLayoutProvider.php

```diff
@@ -18,75 +27,116 @@
 class LtiViewLayoutProvider extends AbstractModificationProvider implements ModificationProvider
 {
 
-    /**
-     * @inheritDoc
-     */
-     
     public function isInterestedInContexts() : ContextCollection
     {
         return $this->context_collection->lti();
     }
-    
+
     /**
-     * This is a basic wip implementation for leaving the lti session with an exit button
-     * 
-     * There is an ongoing discussion about a HeaderInfo bar for different ILIAS modes:
-     * 
-     * https://docu.ilias.de/goto_docu_wiki_wpage_5979_1357.html
-     * 
-     * and a PR Feature Discussion:
-     * 
-     * https://github.com/ILIAS-eLearning/ILIAS/pull/2251
-     * 
+     * @inheritDoc
      */
-    public function getMetaBarModification(CalledContexts $screen_context_stack) : ?MetaBarModification
+
+    public function getPageBuilderDecorator(CalledContexts $screen_context_stack) : ?PageBuilderModification
     {
-        $this->dic->logger()->lti()->info("isActive: " . $this->dic["lti"]->isActive());
-        
+        $this->dic->logger()->lti()->info("getPageBuilderDecorator");
         if ($this->dic["lti"]->isActive()) {
             if (isset($_SESSION['lti_launch_css_url']) && $_SESSION['lti_launch_css_url'] != "") {
                 $this->globalScreen()->layout()->meta()->addCss($_SESSION['lti_launch_css_url']);
             }
-            return $this->globalScreen()
-                ->layout()
-                ->factory()
-                ->metabar()
-                ->withModification(function (MetaBar $current) : ?MetaBar {
-                $f = $this->dic->ui()->factory();
-                $close = $f->button()->close();
-                $exit_symbol = $f->symbol()->glyph()->remove();
-                $exit = $f->button()->bulky($exit_symbol,"exit",$this->dic["lti"]->getCmdLink('exit'));
-                $metabar = $f->mainControls()->metaBar()->withAdditionalEntry('exit', $exit);
-                return $metabar;
-            })->withHighPriority();
-        }
-        else {
-            return null;
+            $this->globalScreen()->layout()->meta()->addCss('./Services/LTI/templates/default/lti.css');
+            $mv = ilMemberViewSettings::getInstance();
+            $isMemberView = false;
+            $ref_id = "";
+            $url = "";
+            if ($mv->isActive()) {
+                $this->dic->logger()->lti()->info("memberView isActive in LTI Mode");
+                $isMemberView = true;
+                $ref_id = $mv->getCurrentRefId();
+                $url = new URI(ilLink::_getLink(
+                    $ref_id,
+                    ilObject::_lookupType(ilObject::_lookupObjId($ref_id)),
+                    array('mv' => 0)
+                ));
+            }
+            return $this->factory->page()->withHighPriority()->withModification(
```

Comment:
do I understand correctly that the problem here is that you want to replace the page completely, but an activated MemberView gets lost?

Reply:
No. with this implementation a course-admin will be able to activate and deactivate the memberview (ModeInfo) within any LTI sessions in the same way as a normal ILIAS course-admin session. That is exactly what we want in LTI :-)